test/timing.py

```diff
@@ -0,0 +1,87 @@
+import subprocess
+import re
+import csv
+import random
+import os
+import time
+import string
+
+def timeChecksum(method):
+    #create the process into a pipe
+    process = subprocess.Popen(['bash', '-c', 'time ' + method],
+                     stdin=open('data.csv', 'r'),
+                     stdout=subprocess.PIPE, 
+                     stderr=subprocess.PIPE,
+                     universal_newlines=True)
+    stdout, stderr = process.communicate() #run the process
+    realtime = re.search("real\\t(.*?)\\n",stderr).group(0)[5:-1] #get the real time
+    m=re.search("(.*?m)",realtime).group(0)
+    m=m[:len(m)-1]
+    s=re.search("m(.*?)s",realtime).group(0)
+    s=s[1:len(s)-1]
+    total_seconds = float(m)*60 + float(s)
+    return total_seconds
+
+#open and write random data to data.csv given the number of rows and cols
+def writeData(row, col):
+    with open('data.csv', mode='w+') as data:
+        data_writer = csv.writer(data, delimiter=',', quotechar='"', quoting=csv.QUOTE_MINIMAL)
+        cols = []
+        for c in range(0,col):
+            cols.append("col"+str(c))
+        data_writer.writerow(cols)
+        for r in range(0,row):
+            row = []
+            for c in range(0,col):
+                if(c%5 == 0):
```

Comment:
this is more frequent than we'd expect in most data sets for "large text". 1 or 2 columns, tops, not 20% of them.

Reply:
changed to include "large text" in column 0 (in all data sets) and column 2 if there are more than 5 cols.